File: Tut3_ROS_Service/src/add_two_int/scripts/add_two_ints_client.py
#!/usr/bin/env python3

import rospy
from rospy_tutorials.srv import AddTwoInts

if __name__ == '__main__':

    #select service name
    nodeName = "add_two_ints_client"

    #select service name
    serviceName = "/add_two_ints"

    #Init Node with its name
    rospy.init_node(nodeName)

    #Send a Log msg in terminal
    rospy.loginfo(nodeName+" has been created")

    #wait for the service
    rospy.wait_for_service(serviceName)

    try:
        add_two_ints = rospy.ServiceProxy(serviceName, AddTwoInts)
        response = add_two_ints(5,-1)
        rospy.loginfo("Sum is: " + str(response.sum))
    except rospy.ServiceException as e:
        rospy.logwarn("Service Failed " + str(e))

File: Tut3_ROS_Service/src/add_two_int/src/add_two_ints_client.cpp
#include <ros/ros.h>
#include <string>
#include <rospy_tutorials/AddTwoInts.h>


int main(int argc, char **argv)
{
    //select Ros node name
    const std::string nodeName("add_two_ints_client_roscpp");

    //select Ros topic name
    const std::string serviceName("/add_two_ints_server");

    //Init Node with its name
    ros::init(argc, argv, nodeName);

    //create node handle
    ros::NodeHandle nh;

    //Configure node node type(publisher) , the topic Name, and queue size
    ros::ServiceClient client(nh.serviceClient<rospy_tutorials::AddTwoInts>(serviceName));

    //Select the rate of node execution
    rospy_tutorials::AddTwoInts srv;
    srv.request.a = 12;
    srv.request.b = 5;

    if(client.call(srv))
    {
        const std::string sum("The Sum of"+ std::to_string(srv.request.a) + " and " + std::to_string(srv.request.b) 
                              + " ---> "+ std::to_string(srv.response.sum));
        ROS_INFO(sum.c_str());
    }
    else
    {
        ROS_WARN("Service call failed");
    }
}
